### DISCUSSION TITLE
Rounding errors causing "increasing" values to not be increasing

### DISCUSSION THREAD

**Discussion by LucidityCrash** [hultenvp/solis-sensor#282] [General]
Dontcha just love the rounding errors on the API (ignore the current unknown I just restarted HA)

![image](https://user-images.githubusercontent.com/49124523/233860186-4cf35411-10fc-4260-81c9-9d5597fe5734.png)

![image](https://user-images.githubusercontent.com/49124523/233860193-610bbba9-7067-45ac-9b27-b5560821327b.png)

![image](https://user-images.githubusercontent.com/49124523/233860221-de303dff-80d5-4ea4-b1db-66ed26c0cbcb.png)

**Comment by astronaut63** [comment]
I was seeing something similar with many of the total values only increasing in steps of 10 - which was then messing up some hourly/daily weekly meters.  In this figure the TotalPanelPower is being pulled directly from my inverter using a Rest call,  Solis Energy total is coming from this integration.  I fixed this locally by editing soliscloud_api.py  for the affected parameters and changing the number of decimals from 2 to 3  e.g. INVERTER_ENERGY_THIS_YEAR:        ['eYear', float, 3],#changed from 2.   I haven't proposed this a change as I wasn't sure  if it would work for all cases e..g does total energy at higer levels start being reported as MWh - and then 6 dp would be needed to get 1 KWh resolution?
![image](https://github.com/hultenvp/solis-sensor/assets/84227537/16e5a660-0259-47f7-acc8-71357bb06241)

**Reply by hultenvp** [reply]
Created issue #297.
Energy indeed does start to be reported in MWh at higher levels, but HA cannot deal with variable units, so converted back to kWh by the integration.
